### ISSUE TITLE
Unified highlighter in word scanning mode does not return proper fragments

### ISSUE THREAD

**Issue by jacool** [elastic/elasticsearch#30136]
**Elasticsearch version** (`bin/elasticsearch --version`):
6.2.3

**Plugins installed**: []

**JVM version** (`java -version`):
openjdk version "1.8.0_161"

**OS version** (`uname -a` if on a Unix-like system):
Linux 5137c3a21142 4.9.87-linuxkit-aufs

**Description of the problem including expected versus actual behavior**:
The unified highlighter used with **"boundary_scanner": "word"** option produces only the searched words without any context disregarding the specified fragment_size.

**Steps to reproduce**:
```
PUT /test
{
  "mappings": {
    "t": {
      "properties": {
        "message": {
          "type": "text",
          "term_vector": "with_positions_offsets"
        }
      }
    }
  }
}

POST /test/t/1
{
    "message": "Sir Tristram, violer d’amores, fr’over the short sea, had passen-core rearrived from North Armorica on this side the scraggy isthmus of Europe Minor to wielderfight his penisolate war: nor had topsawyer’s rocks by the stream Oconee exaggerated themselse to Laurens County’s gorgios while they went doublin their mumper all the time: nor avoice from afire bellowsed mishe mishe to tauftauf thuartpeatrick not yet, though venissoon after, had a kidscad buttended a bland old isaac: not yet, though all’s fair in vanessy, were sosie sesthers wroth with twone nathandjoe. Rot a peck of pa’s malt had Jhem or Shen brewed by arclight and rory end to the regginbrow was to be seen ringsome on the aquaface."
}

GET /test/_search
{
  "version": true,
  "query": {
    "simple_query_string": {
      "fields": [ "message" ],
      "query": "nor"
    }
  },
  "highlight": {
    "fields": {
      "message": {
        "type": "unified",
        "boundary_scanner": "word",
        "fragment_size": 200
      }
    }
  }
}
```

Result:

```
"highlight": {
     "message": [
        "<em>nor</em>",
        "<em>nor</em>"
     ]
 }
```

**Comment by romseygeek** [comment]
I think this is just how the highlighter works?  If you pass a word breakiterator, then the passages returned are individual words.  `fragment_size` is an upper bound, not a lower one.  You probably want to use a sentence breakiterator instead.

**Comment by jacool** [comment]
@romseygeek The very purpose of a highlighter is to bring context to the found words - for the user knows what she searched for. You may learn that from observing how FVH works or how Unified works in the "sentence" mode. All other highlighters/modes will return the maximum number of "constituents" matching the requested fragment_size. E.g. in the sentence mode as many sentences will be returned in a fragment as required to fill the requested fragment_size. The mode (char/word/sentence) only dictates where the fragment braking is allowed to occur.

**Comment by jimczi** [comment]
I agree, we should be able to decorrelate the fragment size and the fragment breaking. The fact that we return as many sentences as required to fill the requested fragment_size is new in 6.2 and it's not applied to the `word` fragmenter. Though I am not sure we need both fragmenter, why would you use the `word` fragmenter instead of the `sentence` one and what do you expect in terms of context for the `word` fragmenter ?

**Comment by mayya-sharipova** [comment]
Still a bug in Elasticsearch 8.12
